### DISCUSSION TITLE
Compatibility with Brewfile

### DISCUSSION THREAD

**Discussion by aboustayyef** [milanvarady/Applite#33] [Ideas]
Hey... I really like this app.

I noticed Applite knows which apps I already installed using brew cask, but I'm not sure if it does this by reading my Brewfile.
It would really be cool to have my Brewfile as where the Applite state is stored. Is that a possible way to go for you?

**Comment by milanvarady** [comment]
Applite gets the currently installed casks by running `brew list --cask` on the selected brew executable. I haven't used Brewfiles so far to be honest, can you elaborate a bit on what you mean by having your Brewfile where the Applite state is?

**Reply by aboustayyef** [reply]
Hi,

a Brewfile is essentially a list of of all your brew packages in one dotfile that can all be installed using one command: `brew bundle install`. Great for when you get a new mac. The idea is that you always keep this list up to date with the apps and packages you use. I was first [introduced to the idea by Casey Liss here](https://www.caseyliss.com/2019/10/8/brew-bundle). It's a great read if you want to know more about it.

So my idea is that Applite can function as some form as a Brewfile editor. Whenever you add new apps, they are added to the casks list, and when apps are uninstalled, they are removed. And when Applite is first installed, it can look at the Brewfile and see which apps haven't been downloaded yet and offer to download them... just spitballing here, but I think something can be done there... 

In any case, thanks for the great app. Very useful.

**Reply by milanvarady** [reply]
I have already implemented basic import/export functionality. It doesn't use a Brewfile, just a txt file of installed apps. 
Correct me if I'm wrong, but I don't see why updating the Brewfile on every app install/uninstall would be beneficial. I could extend the current import/export functionality to work with Brewfiles. That way, you could go in and export all your casks at any time with a push of a button.

**Comment by milanvarady** [comment]
With the v1.2 update, Brewfile import/export is now possible. I'm closing this discussion for now.
